docs/import/manage_disaster_flexible.js

```diff
@@ -1,22 +1,154 @@
+ * action. Thus, each select whose value cascades (poverty, geography, damage,
+ * buildings) must have a `PendingChecker`. Similarly initialization as a whole
+ * has a `PendingChecker`. This keeps our count of pending operations limited to
+ * the types of operations there can be, which is correct: the user can never be
+ * waiting on more than one operation of a given type.
+ *
+ * A subtle issue arises when a pending value becomes unnecessary: for instance,
+ * if the div that needs to be populated is hidden. Then we immediately count
+ * down the pending operations via {@link markInvisible}. But if the div ever
+ * becomes visible again, and the operation has still not finished, we count
+ * pending operations back up using {@link markVisible}. This allows listings to
+ * complete in the background and be ready, even if the user is switching other
+ * options back and forth.
+ */
+class PendingChecker {
+  pending = PendingState.NOT_PENDING;
+
+  /** @return {boolean} False if this checker was already pending */
+  maybeStartPending() {
+    if (!isFlexible()) {
+      return true;
+    }
+    switch (this.pending) {
+      case PendingState.NOT_PENDING:
+        this.pending = PendingState.PENDING;
+        startPendingOperation();
+        return true;
+      case PendingState.PENDING:
+        return false;
+      case PendingState.PENDING_BUT_INVISIBLE:
+        showError('Error validating inputs: please reload page');
+        return false;
+    }
+  }
+
+  /**
+   * Notes that the data source this tracks has finished its pending operation.
+   * Must only be called if the data is still valid. Should be called after
+   * display of the data is done (so, at end of {@link setOptionsForColumns}).
+   */
+  finishPending() {
+    if (!isFlexible()) {
+      return;
+    }
+    switch (this.pending) {
+      case PendingState.NOT_PENDING:
+        showError('Error validating inputs: please reload page');
+        return;
+      case PendingState.PENDING:
+        countDownPendingOperations();
+        // Fall through.
+      case PendingState.PENDING_BUT_INVISIBLE:
+        this.pending = PendingState.NOT_PENDING;
+    }
+  }
+
+  /** See class doc. */
+  markInvisible() {
+    if (this.pending === PendingState.PENDING) {
+      this.pending = PendingState.PENDING_BUT_INVISIBLE;
+      countDownPendingOperations();
+    }
+  }
+
+  /** See class doc. */
+  markVisible() {
+    if (this.pending === PendingState.PENDING_BUT_INVISIBLE) {
+      this.pending = PendingState.PENDING;
+      startPendingOperation();
+    }
+  }
+}
 
 /**
  * Functions for dealing with a "flexible" (non-state-based) disaster.
  *
+ * ## Structure of a flexible disaster
+ *
+ * A flexible disaster takes more general input assets than a state-based one.
+ * We expect the following (analogs to the state-based case are in parentheses):
+ *
+ * 1. A poverty asset, which must have the following columns:
+ *      a. Poverty rate ('SNAP PERCENTAGE');
```

Comment:
nit maybe prepend these examples with e.g. to make it clear it doesn't need to be those exact cols